src/management.jl

```diff
@@ -0,0 +1,79 @@
+function get_backtrace(ex)
+    v = IOBuffer()
+    Base.showerror(v, ex, catch_backtrace())
+    return v.data |> copy |> String
+end
+
+function get_log_level(l)
+    log_level_str = lowercase(l)
+
+    (log_level_str == "debug") ? Logging.Debug :
+    (log_level_str == "info")  ? Logging.Info  :
+    (log_level_str == "warn")  ? Logging.Warn  : Logging.Error
+end
+
+function status_monitor(stop_file::AbstractString,
+                        zsock::MessageSocket,
+                    )
+    while isopen(zsock.sock)
+        sleep(5)
+        flush(stdout); flush(stderr);
+        # stop server if stop is requested
+        if isfile(stop_file)
+            @warn("Server stop requested.")
+            flush(stdout); flush(stderr)
+
+            # stop accepting new requests
+            close(zsock.sock)
+
+            rm(stop_file; force=true)
+        end
+    end
+end
+
+function status_monitor(stop_file::AbstractString,
+                        event_queue::Channel,
+                        httpsock::Ref{Sockets.TCPServer},
+                    )
+    while isopen(event_queue)
+        sleep(5)
+        flush(stdout); flush(stderr);
+        # stop server if stop is requested
+        if isfile(stop_file)
+            @warn("Server stop requested.")
+            flush(stdout); flush(stderr)
+
+            # stop accepting new requests
+            close(httpsock[])
+
+            # wait for queued requests to be processed and close queue
+            while isready(event_queue)
+                yield()
+            end
+            close(event_queue)
+
+            rm(stop_file; force=true)
+        end
+    end
+end
+
+function recover(name::AbstractString, keep_running::Function,
+                 do_action::Function, handle_exception::Function; backoff=0, backoffmax=120, backoffincrement=1)
+    while keep_running()
+        try
+            do_action()
+            backoff = 0
+        catch ex
+            exception_action = handle_exception(ex)
+            if exception_action == :exit
+                @warn("Stopping", name)
+                return
+            else # exception_action == :continue
+                bt = get_backtrace(ex)
+                @error("Recovering from unknown exception", name, ex, bt, backoff)
```

Comment:
won't passing the `exception` kwarg to `@error` cause a backtrace to b e shown?
Or is that only in the `ConsoleLogger` ?

Reply:
It does print the backtrace. But its all on one line. Difficult to read.